PR Template Rough Draft

cprakti opened the pull request:
Ironically... this PR description is garbage. But please leave comments on the template.

cprakti committed:
PR Template Rough Draft

Ironically... this PR description is garbage. But please leave comments on the template.

pnewsam commented:
I like it! Very thorough. I would vote for cutting out these sections:

- Description - Potential side effects
- Description - Additional context or questions for reviewer
- Testing - Browser list
- Checklist - I have reviewed the PR myself.
- Checklist - I have updated the JIRA ticket with a link to this PR.
- Checklist - I have rebased off the most recent version of the master branch.
- Checklist - I have messaged the team about any breaking changes, new ENV variables, special deployment instructions, etc.

Just because I think people would be more likely to follow it (and not forget/skip over things) if it's short and sweet.

pnewsam commented on PR_TEMPLATE.md:
```diff
@@ -0,0 +1,45 @@
+   - [ ] Edge
+   - [ ] N/A
+
+## Checklist
```
How would you feel about condensing these phrases? Ie. "Written tests" instead of "I have written tests that fail without this code."

cprakti replied:
Yep, works for me.

cprakti commented:
> I like it! Very thorough. I would vote for cutting out these sections:
> 
> * Description - Potential side effects
> * Description - Additional context or questions for reviewer
> * Testing - Browser list
> * Checklist - I have reviewed the PR myself.
> * Checklist - I have updated the JIRA ticket with a link to this PR.
> * Checklist - I have rebased off the most recent version of the master branch.
> * Checklist - I have messaged the team about any breaking changes, new ENV variables, special deployment instructions, etc.
> 
> Just because I think people would be more likely to follow it (and not forget/skip over things) if it's short and sweet.

@pnewsam, I think that's a good point. I'd rather have something people think is useful, than some unpractical tome. I opted for adding in everything I could thing of, assuming we'd pare it back.

I think the bigger question this raises is what is optional vs. mandatory. Do you think it's better to ask everyone to include all sections or just have them available but optional.

pnewsam commented:
Hmm that's true we could definitely remove any sections later on if they're underused if they're underused. I'm totally happy leaving them in and iterating.
